kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -62,6 +66,29 @@ public LogSegment(
       List<FileStatus> checkpoints,
       Optional<Long> checkpointVersionOpt,
       long lastCommitTimestamp) {
+    requireNonNull(logPath, "logPath is null");
+    requireNonNull(deltas, "deltas is null");
+    requireNonNull(checkpoints, "checkpoints is null");
+    requireNonNull(checkpointVersionOpt, "checkpointVersionOpt is null");
+    checkArgument(
+        checkpoints.isEmpty() == !checkpointVersionOpt.isPresent(),
+        "checkpoints and checkpointVersionOpt must either be both empty or both non-empty");
```

Comment:
it's not possible to hit this in the code (in theory) because we thrown an error during log segment building in this case right?